File: PSet1 (C)/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // US Coins
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    int coins = 0;
    float change = 0.0;

    change = get_float("Change owed: \n");  //Ask user, how much change is owed
    int ch_round = round(change * 100);

    while (change <= 0) // Check if user input is valid
    {    
        change = get_float("Change owed: \n");
    }

    // Check how many Coins we need
    for (int i = ch_round; i >= quarter; i = (i - quarter)) // Check how many quarters we can use
    {
        ch_round = ch_round - quarter;
        coins++;
    }

    for (int i = ch_round; i >= dime; i = (i - dime))    // Check how many dimes we can use
    {
        ch_round = ch_round - dime;
        coins++;
    }

    for (int i = ch_round; i >= nickel; i = (i - nickel))    // Check how many nickels we can use
    {
        ch_round = ch_round - nickel;
        coins++;
    }

    for (int i = ch_round; i >= penny; i = (i - penny))  // Check how many pennys we can use
    {
        ch_round = ch_round - penny;
        coins++;
    }
    printf("%d\n", coins);
}

File: PSet6 (Python)/cash/cash.py
# Import get_float from the cs50 libary
from cs50 import get_float

# US coins
quarter = 25
dime = 10
nickel = 5
penny = 1

coins = 0
change = 0.00

# Ask user, how much change is owed
while change <= 0:
    change = get_float("Change owed: ")

# Round change so we get an int
ch_round = round(change * 100)

# Check how many "quarters" we can use
while ch_round >= quarter:
    ch_round -= quarter
    coins += 1

# Check how many "dimes" we can use
while ch_round >= dime:
    ch_round -= dime
    coins += 1

# Check how many "nickels" we can use
while ch_round >= nickel:
    ch_round -= nickel
    coins += 1

# Check how many "pennys" we can use
while ch_round >= penny:
    ch_round -= penny
    coins += 1

print(coins)